client/components/language-picker/modal.jsx

```diff
@@ -324,4 +328,5 @@ export class LanguagePickerModal extends PureComponent {
 
 export default connect( state => ( {
 	localizedLanguageNames: getLocalizedLanguageNames( state ),
+	currentLocaleSlug: getLocaleSlug(),
```

Comment:
Well, I'm having a complicated feeling about this :)
The benefit of using `connect()` from `react-redux` to wire redux substate to react props is that updates of the substate will trigger re-rendering of the component. However, `getLocaleSlug()` here has nothing to do with the underlying redux state, so actually it will never trigger re-rendering because of this `connect()` call. One quick way to verify this is to:
1. Expose the `switchLocale` function and `getLocaleSlug` to the console.
e.g. 
```
if ( typeof window !== 'undefined' ) { 
    window.switchLocale = switchLocale;
    window.getLocaleSlug = getLocaleSlug;
}
```
1. Open the picker modal window.
1. Run `switchLocale( 'any slug you want' )`, and see that the modal window won't update even if the output of `getLocaleSlug()` has changed.

However, I have seen many other places are doing this and I can see why it is charming in some cases. Thus I don't hold a strong opinion on it, just wanted to mention it :)